backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/MoveDiskCommand.java

```diff
@@ -170,11 +170,15 @@ protected Map<String, Pair<String, String>> getExclusiveLocks() {
 
     private String getDiskIsBeingMigratedMessage() {
         if (cachedDiskIsBeingMigratedMessage == null) {
+            DiskImage diskImage = diskImageDao.get(getParameters().getImageId());
+            String diskName = "";
+            if (diskImage != null) {
+                diskName = diskImage.getDiskAlias() != null ?  diskImage.getDiskAlias() : "";
+            }
             cachedDiskIsBeingMigratedMessage = new LockMessage(EngineMessage.ACTION_TYPE_FAILED_DISK_IS_BEING_MIGRATED)
-                    .withOptional("DiskName", getParameters().getNewAlias())
+                    .withOptional("DiskName", diskName)
```

Comment:
Sorry for my previous comment, I misunderstood the issue and the behavior of `withOptional()`.
After some checking I see:
1. Since `diskName` is initialized to empty `String`, no need to use ternary operator in line 176. Actually a much simple code could be to change lines 175-177 to:
```
if (diskImage != null && diskImage.getDiskAlias() != null) {
    diskName = diskImage.getDiskAlias();
}
```
2. After your checks, you don't need to use `withOptional()` at all, you can use a simple `with()`, since the `diskName` value will never be `null`.
3. Actually @ahadas I don't see any good usage for `withOptional()` at all! Looks like the same issue as in this bug (with unparsed key) happens in its every usage. The better solution could be to change `withOptional()` to replace the key with `""` (or `StringUtils.EMPTY`), rather than just doing nothing and maybe even add a new `withOptional()` version that receives a default value (to allow passing values like `N/A`).

Reply:
Maybe touching 16 files as one of your first PRs is scary :) but IMHO @ahadas worth considering option 3 above.
This could also simplify https://github.com/oVirt/ovirt-engine/pull/383/ (that was already merged) as well.
I don't understand what was the intended use case of current `withOptional()` implementation, but looks like issue of this PR happens everywhere the current implementation of `withOptional()`is used and could result in messages like the following to be displayed to the user:
`Cannot remove VM-Pool. VM Pool ${VmPoolName} is being removed with VM ${VmName}`.
While probably in this case it could be (though not perfect as well) to replace both `${VmPoolName}` & `${VmName}` with either empty `String` or "N/A".
And in https://github.com/oVirt/ovirt-engine/pull/383/ , where currently (after `null` checking) you could use a simple `with()`, the whole check could go into a new `withOptional()` that automatically replaces `null` with empty `String`.
Just something to consider, aka "Yeah, well, you know that's just like, uh, your opinion man" - The Big Lebowski :)

Reply:
@barpavel from my perspective there's just a minor benefit in having empty string rather than 'null' - the issue here is less about that part and more about the fact we really should have the disk alias mentioned there